core/src/main/java/nu/yona/server/subscriptions/service/UserDTO.java

```diff
@@ -154,17 +154,12 @@ static UserDTO createInstance(User userEntity)
 	{
 		if (userEntity == null)
 		{
-			return createRemovedUserInstance();
+			return null;
```

Comment:
There were 4 callers that needed the `null` check. I have created an alternative method `createInstanceIfNotNull` that returns `null` or an instance and added some comments to the methods to explain it.